pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -54,19 +55,31 @@ var _ webhook.Validator = &Cluster{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Cluster) ValidateCreate() error {
 	clusterlog.Info("validate create", "name", r.Name)
-	if r.IsReconcilePaused() {
-		clusterlog.Info("cluster is paused, so allowing create", "name", r.Name)
-		return nil
-	}
+	var allErrs []error
+
 	if !features.IsActive(features.FullLifecycleAPI()) {
-		return apierrors.NewBadRequest("Creating new cluster on existing cluster is not supported")
+		allErrs = append(allErrs,
+			fmt.Errorf("creating new cluster on existing cluster is not supported"))
 	}
+
+	if err := r.Validate(); err != nil {
+		allErrs = append(
+			allErrs,
+			fmt.Errorf("invalid cluster %v", err.Error()))
+	}
+
 	if r.IsSelfManaged() {
-		return apierrors.NewBadRequest("Creating new cluster on existing cluster is not supported")
+		if !r.IsReconcilePaused() {
+			allErrs = append(
+				allErrs, fmt.Errorf("creating new cluster on existing cluster is not supported for self managed clusters"))
+		} else {
+			clusterlog.Info("cluster is paused, so allowing create", "name", r.Name)
+		}
 	}
 
-	if err := validateCNIPlugin(r.Spec.ClusterNetwork); err != nil {
-		return apierrors.NewBadRequest(err.Error())
+	if len(allErrs) > 0 {
+		aggregate := utilerrors.NewAggregate(allErrs)
+		return apierrors.NewBadRequest(fmt.Sprintf("validating cluster spec: %v", aggregate))
```

Comment:
do we need another top level message? I would just use the error from the aggregate directly

Reply:
Also, it might be a good idea to look at how StatusErrors are build from field.ErrorList, I think that process can help us do the same for `[]error`